### PR TITLE
Fix NulRefEx from callback and add wcf callback behavior error handling

### PR DESCRIPTION
**Pull request by ericdevg**
#576

@AndreSayreMSFT @HuanhuanSunMSFT @Microsoft/poshtools

### CODE REVIEW

File: PowerShellTools/DebugServiceEventHandlerBehaviorAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+namespace PowerShellTools
+{
+    /// <summary>
+    /// In WCF, unhandled exception crashes the service, leaving the channel into fault state, which is basically requiring client to re-instantiate proxy in order to continue using the service.
+    /// As a result, WCF provides the ability to configure a service to return information from unhandled exceptions.
+    /// This is implemetaion of the generic error handler for entire powershell wcf services, by exposing it as a service behavior attribute, so that it is developer friendly
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    class DebugServiceEventHandlerBehaviorAttribute : Attribute, IErrorHandler, IEndpointBehavior
```
Comment (ASayre) [inline PowerShellTools/DebugServiceEventHandlerBehaviorAttribute.cs:21]: access modifier

Comment (ericdevg) [inline PowerShellTools/DebugServiceEventHandlerBehaviorAttribute.cs:21]: Fixed.

File: PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs [hunk 2]
```diff
@@ -209,7 +216,14 @@ public void ClearHostScreen()
...
         {
-            return Debugger.ReplWindow.WaitKey();
+            if (Debugger.ReplWindow != null)
+            {
+                return Debugger.ReplWindow.WaitKey();
+            }
+            else
+            {
+                return null;
+            }
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs:226]: Confirm that in this case we want the PSRawHost to throw a PipelineStoppedException?

**Comment by ASayre** [comment]
looks good!

File: PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs (hunk 2 above)
Comment (ericdevg) [inline PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs:226]: yes, that is right, which stops current pending readkey request.

**Pull request merged**